assets/js/components/KeyMetrics/ChipTabGroup/index.js

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect } from 'googlesitekit-data';
+import {
+	KEY_METRICS_CURRENT_SELECTION_GROUP_SLUG,
+	KEY_METRICS_GROUPS,
+	KEY_METRICS_SELECTED,
+	KEY_METRICS_SELECTION_FORM,
+} from '../constants';
+import { CORE_FORMS } from '../../../googlesitekit/datastore/forms/constants';
+import Chip from './Chip';
+import MetricItem from '../MetricsSelectionPanel/MetricItem';
+import NoSelectedItemsSVG from '../../../../svg/graphics/key-metrics-no-selected-items.svg';
+import {
+	BREAKPOINT_SMALL,
+	BREAKPOINT_TABLET,
+	useBreakpoint,
+} from '../../../hooks/useBreakpoint';
+
+export default function ChipTabGroup( { allMetricItems, savedItemSlugs } ) {
+	const [ isActive, setIsActive ] = useState(
+		KEY_METRICS_CURRENT_SELECTION_GROUP_SLUG
+	);
+	const breakpoint = useBreakpoint();
+	const isDesktopBreakpoint =
+		breakpoint !== BREAKPOINT_SMALL && breakpoint !== BREAKPOINT_TABLET;
+
+	const selectedMetrics = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue(
+			KEY_METRICS_SELECTION_FORM,
+			KEY_METRICS_SELECTED
+		)
+	);
+
+	const currentSelectionGroup = {
+		SLUG: KEY_METRICS_CURRENT_SELECTION_GROUP_SLUG,
+		LABEL: __( 'Current selection', 'google-site-kit' ),
+	};
+	const selectedCounts = {};
+	const activeMetricItems = {};
+
+	for ( const metricItemSlug in allMetricItems ) {
+		const metricGroup = allMetricItems[ metricItemSlug ].group;
+		if ( metricGroup === isActive ) {
+			activeMetricItems[ metricItemSlug ] =
+				allMetricItems[ metricItemSlug ];
+		}
+
+		if ( ! selectedCounts[ metricGroup ] ) {
+			const selectedCount = Object.keys( allMetricItems ).filter(
+				( slug ) => {
+					// Check if metric slug is in selectedMetrics, so the group
+					// count is reflected in real time as metrics are checked/unchecked.
+					if (
+						allMetricItems[ slug ].group === metricGroup &&
+						selectedMetrics.includes( slug )
+					) {
+						return true;
+					}
+
+					return false;
+				}
+			).length;
+			selectedCounts[ metricGroup ] = selectedCount;
+		}
+		// @TODO Will be updated in #9385 for current selection.
+		selectedCounts[ KEY_METRICS_CURRENT_SELECTION_GROUP_SLUG ] = 0;
+	}
+
+	const onChipChange = useCallback(
+		( slug ) => {
+			setIsActive( slug );
+		},
+		[ setIsActive ]
+	);
+
+	const chipItemRows = [
+		[ currentSelectionGroup, ...KEY_METRICS_GROUPS.slice( 0, 2 ) ],
+		[ ...KEY_METRICS_GROUPS.slice( 2 ) ],
+	];
+
+	return (
+		<div className="googlesitekit-chip-tab-group">
+			<div className="googlesitekit-chip-tab-group__tab-items">
+				{ isDesktopBreakpoint &&
```

Comment:
Since layout is a bit atypical, common grid/column won't work, as it will be evenly distributed in width, and in this case all items keep flexible width within their rows, and last 2 items are having a bit bigger padding, so I split this into 2 rows for non-mobile screens, and kept it in single row for mobile, to ensure proper layout flow